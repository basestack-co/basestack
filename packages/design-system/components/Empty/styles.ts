import styled from "styled-components";
import { rem } from "polished";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.empty.backgroundColor};
  border-radius: ${rem("6px")};
  padding: ${({ theme }) => theme.spacing.s8} ${({ theme }) => theme.spacing.s5};
  border-radius: ${rem("6px")};
  justify-content: center;
  align-items: center;
`;

export const IconContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: ${rem("48px")};
  width: ${rem("48px")};
  border-radius: ${rem("8px")};
  background-color: ${({ theme }) => theme.empty.icon.backgroundColor};
  margin-bottom: ${({ theme }) => theme.spacing.s3};
`;
