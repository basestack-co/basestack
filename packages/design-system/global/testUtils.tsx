import { type RenderOptions, render } from "@testing-library/react";
import type React from "react";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "../theme/lightTheme";

interface Props {
  children: React.ReactNode;
}

const WithThemeProvider = ({ children }: Props) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithTheme };
