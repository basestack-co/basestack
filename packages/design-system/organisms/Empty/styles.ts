import styled from "styled-components";
import { rem } from "polished";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.colors.gray100};
  border-radius: ${rem("6px")};
  padding: ${({ theme }) => theme.spacing.s8} ${({ theme }) => theme.spacing.s5};
  border-radius: ${rem("6px")};
  justify-content: center;
  align-items: center;
`;

export const IconContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: ${rem("48px")};
  width: ${rem("48px")};
  border-radius: ${rem("8px")};
  background-color: ${({ theme }) => theme.colors.blue100};
  margin-bottom: ${({ theme }) => theme.spacing.s3};
`;
