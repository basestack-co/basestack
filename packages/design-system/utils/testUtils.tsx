import defaultTheme from "@basestack/design-system/theme/lightTheme";
import isPropValid from "@emotion/is-prop-valid";
import { RenderResult, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

const Provider = ({
  children,
  theme = defaultTheme,
}: {
  children: React.ReactNode;
  theme: any;
}) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

export const renderWithTheme = (
  ui: React.ReactElement,
  options?: any,
): RenderResult => render(ui, { wrapper: Provider, ...options });
