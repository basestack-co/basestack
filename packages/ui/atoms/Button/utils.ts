import { Size, Target, Type, Variant } from "@/components/atoms/Button/types";
import { css, DefaultTheme } from "styled-components";
import { darken, rem, transparentize } from "polished";

export const handleButtonSize = (size?: Size) => {
  switch (size) {
    case "large":
      return css`
        height: ${rem("60px")};
        font-size: ${rem("18px")};
        line-height: ${rem("25px")};
        padding: 0 ${rem("40px")};
      `;
    case "medium":
      return css`
        height: ${rem("54px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("34px")};
      `;
    case "normal":
      return css`
        height: ${rem("48px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("30px")};
      `;
    default:
    case "small":
      return css`
        height: ${rem("46px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("24px")};
      `;
    case "xSmall":
      return css`
        height: ${rem("42px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("20px")};
      `;
  }
};

export const handleButtonsColors = (theme: DefaultTheme) => ({
  primary: {
    outline: {
      color: theme.colors.primary,
      borderColor: theme.colors.primary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.primary,
    },
  },
  secondary: {
    outline: {
      color: theme.colors.secondary,
      borderColor: theme.colors.secondary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.secondary,
    },
  },
  tertiary: {
    outline: {
      color: theme.colors.gray3,
      borderColor: theme.colors.gray3,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: theme.colors.gray6,
    },
  },
  neutral: {
    outline: {
      color: theme.colors.gray2,
      borderColor: theme.colors.gray2,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: "transparent",
    },
  },
  success: {
    outline: {
      color: theme.colors.success,
      borderColor: theme.colors.success,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.success,
    },
  },
  warning: {
    outline: {
      color: theme.colors.warning,
      borderColor: theme.colors.warning,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.warning,
    },
  },
  danger: {
    outline: {
      color: theme.colors.error,
      borderColor: theme.colors.error,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.error,
    },
  },
});

export const darkenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return darken(
    amount || 0.1,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};

export const lightenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return transparentize(
    amount || 0.6,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};
