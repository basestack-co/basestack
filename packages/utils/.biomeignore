# Node dependencies
node_modules

# Next.js build output
.next
out

# Environment files
.env
.env.*
!.env.example

# Logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
pnpm-debug.log*

# OS / editor files
.DS_Store
Thumbs.db

# Package manager cache
.pnp
.pnp.js
.pnp.cjs

# Test coverage
coverage

# Build artifacts
dist
build

# Temporary files
.tmp
temp