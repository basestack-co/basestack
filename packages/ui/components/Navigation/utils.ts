// Types
import { Translate } from "next-translate";
import { NextRouter } from "next/router";
// Utils
import { config } from "@basestack/utils";
import { signOut } from "next-auth/react";

export const getInternalLinks = (t: Translate, formId: string) => {
  return [
    {
      icon: "flag",
      activeText: "Submissions",
      to: `/form/${formId}/submissions`,
      text: t("internal.submissions"),
    },
    {
      icon: "flag",
      activeText: "Setup",
      to: `/form/${formId}/setup`,
      text: t("internal.setup"),
    },
    {
      icon: "settings",
      activeText: "Settings",
      to: `/form/${formId}/settings/general`,
      text: t("internal.settings"),
    },
  ];
};

export const getExternalLinks = (t: Translate) => {
  return [
    {
      icon: "description",
      activeText: "Documentation",
      to: config.urls.docs.flags.base,
      isExternal: true,
      text: t("external.docs"),
    },
    {
      icon: "link",
      activeText: "Github",
      to: config.urls.repo,
      isExternal: true,
      text: t("external.github"),
    },
  ];
};

export const getAvatarDropdownList = (
  t: Translate,
  router: NextRouter,
  onCreateForm: () => void,
) => {
  return [
    {
      icon: "add_circle",
      text: t("create.form"),
      onClick: onCreateForm,
      separator: true,
    },
    {
      icon: "settings",
      text: t("dropdown.settings"),
      onClick: () =>
        router.push({
          pathname: "/user/profile/settings",
        }),
    },
    {
      icon: "logout",
      text: t("dropdown.logout"),
      onClick: signOut,
    },
  ];
};
