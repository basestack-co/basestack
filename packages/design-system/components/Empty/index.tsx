import React, { memo } from "react";
import { useTheme } from "styled-components";
import { SpaceProps } from "styled-system";
// Components
import Text from "../Text";
import { Button } from "../Button";
import IconBox from "../IconBox";
import { Container } from "./styles";

export interface EmptyProps extends SpaceProps {
  title: string;
  description: string;
  button?: {
    text: string;
    onClick: () => void;
  };
  iconName?: string;
}

const Empty = ({
  title,
  description,
  button,
  iconName = "info",
  ...props
}: EmptyProps) => {
  const theme = useTheme();

  return (
    <Container {...props}>
      <IconBox
        backgroundColor={theme.empty.icon.backgroundColor}
        icon={iconName}
        mb={theme.spacing.s5}
      />
      <Text size="large" textAlign="center" mb={theme.spacing.s1}>
        {title}
      </Text>
      <Text size="small" textAlign="center" muted>
        {description}
      </Text>
      {button && (
        <Button mt={theme.spacing.s5} onClick={button.onClick}>
          {button.text}
        </Button>
      )}
    </Container>
  );
};

export default memo(Empty);
