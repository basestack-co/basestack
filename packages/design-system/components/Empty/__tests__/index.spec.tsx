import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import Empty from "..";

describe("Empty tests", () => {
  afterEach(cleanup);

  test("should render Empty correctly", () => {
    const { asFragment } = renderWithTheme(
      <Empty
        title="title"
        button={{ text: "button", onClick: jest.fn() }}
        description="description"
        iconName="info"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
