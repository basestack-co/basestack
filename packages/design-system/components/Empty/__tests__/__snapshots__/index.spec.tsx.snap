// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Empty tests should render Empty correctly 1`] = `
<DocumentFragment>
  .c3 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c4 {
  color: #757575;
  font-size: 1.5rem;
}

.c7 {
  height: 2.25rem;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #276EF1;
  color: #FFFFFF;
  width: normal;
  height: normal;
  margin-top: 1.25rem;
  width: fit-content;
}

.c7:hover:not(:active, :disabled) {
  background-color: #0e55d7;
}

.c7 .material-symbols-sharp,
.c7 .material-symbols-rounded {
  color: inherit;
}

.c7:disabled {
  cursor: default;
  opacity: 0.6;
}

.c1 {
  margin-bottom: 1.25rem;
  position: relative;
  height: 3rem;
  width: 3rem;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(220deg,          #F6F6F6 0%,          #CBCBCB 30%,          #AFAFAF 70%,          #F6F6F6 100%);
}

.c1::before {
  content: "";
  position: absolute;
  inset: 1px;
  background: #EEEEEE;
  border-radius: 0.4375rem;
  z-index: 0;
}

.c2 {
  position: relative;
  display: inline-flex;
  z-index: 1;
}

.c6 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #757575;
  font-family: roboto;
  text-align: center;
}

.c5 {
  font-size: 1.125rem;
  line-height: 1.625rem;
  font-weight: 500;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #000000;
  font-family: roboto;
  text-align: center;
  margin-bottom: 0.25rem;
}

.c0 {
  display: flex;
  flex-direction: column;
  background-color: #EEEEEE;
  border-radius: 0.375rem;
  padding: 3.125rem 1.25rem;
  border-radius: 0.375rem;
  justify-content: center;
  align-items: center;
}

<div
    class="c0"
  >
    <div
      class="c1 icon-box"
      mb="1.25rem"
      variant="outlined"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
          data-testid="icon-container"
        >
          <span
            class="c4 material-symbols-rounded"
            data-testid="icon"
          >
            info
          </span>
        </div>
      </div>
    </div>
    <h3
      class="c5"
      font-family="roboto"
      mb="0.25rem"
    >
      title
    </h3>
    <p
      class="c6"
      font-family="roboto"
    >
      description
    </p>
    <button
      class="c7"
      data-testid="button"
      mt="1.25rem"
      variant="primary"
    >
      <span
        class=""
      >
        button
      </span>
    </button>
  </div>
</DocumentFragment>
`;
